database/sqlite3/sqlite3_test.go

```diff
@@ -181,3 +182,35 @@ func TestMigrateWithDirectoryNameContainsWhitespaces(t *testing.T) {
 	}
 	dt.Test(t, d, []byte("CREATE TABLE t (Qty int, Name string);"))
 }
+
+var pathTests = []struct {
+	in  string
+	out string
+}{
+	// simple valid path, no whitespaces
+	{"sqlite3:///Path/To/A/DB/file.db", "/Path/To/A/DB/file.db"},
+	// simple path, with whitespaces
+	{"sqlite3:///Path To/A DB/file name.db", "/Path To/A DB/file name.db"},
+	// path with whitespaces and query params
+	{"sqlite3:///Path To/A DB/file name.db?aQuery=something&bQuery=else&c=d", "/Path To/A DB/file name.db?aQuery=something&bQuery=else&c=d"},
+	// path with whitespaces and query params (including custom query param that should be filtered out)
+	{"sqlite3:///Path To/A DB/file name.db?aQuery=something&bQuery=else&c=d&x-custom-query-param=scrubbed", "/Path To/A DB/file name.db?aQuery=something&bQuery=else&c=d"},
+
+	// path with % escaped characters
+	{"sqlite3:///Path%20To/A%20DB/file%20name.db", "/Path To/A DB/file name.db"},
+	// path with % escaped characters & escaped query params
+	{"sqlite3:///Path%20To/A%20DB/file%20name.db?aQuery=something%20else&c=d", "/Path To/A DB/file name.db?aQuery=something+else&c=d"},
```

Comment:
I'm not sure if sqlite [will handle](https://sqlite.org/uri.html#uri_format) `+` in the query parameters and I [don't see](https://golang.org/pkg/net/url/#Values.Encode) an option to specify how query parameters are encoded in`net/url`. e.g. an option to use `%20` instead of `+`

Reply:
This I am not too sure about. Although, where its a [valid encoding](https://www.w3schools.com/tags/ref_urlencode.asp), I would imagine that it should be supported or be considered a bug in sqlite?

Reply:
After digging around a bit more, it seems the consensus from StackOverflow is "You should have %20 before the ? and + after."    [Post 1](https://stackoverflow.com/questions/1211229/in-a-url-should-spaces-be-encoded-using-20-or)  [Post 2](https://stackoverflow.com/questions/1634271/url-encoding-the-space-character-or-20)

Reply:
Thanks for researching this!